test/extended/fixtures/test-docker-build-quota.json

```diff
@@ -0,0 +1,45 @@
+{
+  "kind": "Template",
+  "apiVersion": "v1",
+  "metadata": {
+    "name": "docker-build-quota",
+    "creationTimestamp": null,
+    "annotations": {
+      "description": "Defines a docker build with quota.  The build will fail intentionally."
+    }
+  },
+  "objects": [
+    {
+      "kind": "BuildConfig",
+      "apiVersion": "v1",
+      "metadata": {
+        "name": "docker-build-quota",
+        "creationTimestamp": null,
+        "labels": {
+          "name": "docker-build-quota"
+        }
+      },
+      "spec": {
+        "resources": {
+          "limits": {
+            "cpu": "60m",
+            "memory": "200Mi"
+          }
+        },
+        "source": {
+          "dockerfile":"FROM centos:7\nUSER root\nRUN echo -n CGROUP= && cat /proc/self/cgroup\nRUN echo -n MEMORY= && cat /sys/fs/cgroup/memory/memory.limit_in_bytes\nRUN echo -n MEMORYSWAP= && cat /sys/fs/cgroup/memory/memory.memsw.limit_in_bytes\nRUN echo -n QUOTA= && cat /sys/fs/cgroup/cpu,cpuacct/cpu.cfs_quota_us\nRUN echo -n SHARES= && cat /sys/fs/cgroup/cpu,cpuacct/cpu.shares\nRUN echo -n PERIOD= && cat /sys/fs/cgroup/cpu,cpuacct/cpu.cfs_period_us\n# exit 1 causes the docker build to fail which causes docker to show the output of all commands like 'cat'.\nRUN exit 1\n"
```

Comment:
The test would be easier to follow if you have a source of type binary and then call 'oc start-build --from-file=[blah]' with a separate Dockerfile